CMakeLists.txt

```diff
@@ -3477,6 +3477,360 @@ endif (gRPC_BUILD_CODEGEN)
 
 endif (gRPC_BUILD_TESTS)
 
+add_library(grpc++_cronet
+  src/cpp/client/cronet_credentials.cc
```

Comment:
We tried local build with only link against libgrpc_cronet and libgrpc++_cronet and got several undefined symbols errors. Looks like the ssl channel credentials related source code are missing. You may need to add these source files to grpc++_cronet lib:
src/cpp/client/secure_credentials.cc
src/cpp/common/auth_property_iterator.cc
src/cpp/common/secure_auth_context.cc
src/cpp/common/secure_channel_arguments.cc
src/cpp/common/secure_create_auth_context.cc

The following server part credentials source code are missing too:
src/cpp/server/insecure_server_credentials.cc
src/cpp/server/secure_server_credentials.cc

Reply:
If you're using cronet client, why does your code depend on:
1. server credentials. gRPC cronet is client only, why is there a server dependency?
2. secure_credentials.cc. Why is there a dependency on this file? I thought you just needed `CronetChannelCredentials` which is defined in src/cpp/client/cronet_credentials.cc.

Reply:
1. server credentials are used by our grpc unit&integration test. We create local grpc server instance in these tests.Please advise us if there is a better approach in this case case.
2. secure_credentials are used to create SSL secure channel with http2 transport. We want grpc client able to create both ssl secure channel and cronet channel so we can use AB test control to test&rollout cronet channel gradually over ssl secure channel in production. Otherwise, we had to switch between using ssl secure channel and cronet channel at the build time. That's hard for feature ramp-up in production. 
Thanks.

Reply:
Thanks for the explanation. I think we can solve your issues with unresolved symbols, duplicate symbols by packaging all the required code (cronet client, http2 client, server) in libgrpc++_cronet.
Then you would only depend on one library - libgrpc++_cronet.
I have made made this change - PTAL.

Reply:
You can experiment with making changes to the lib dependencies yourself. Here's how it works:

[CMakelists.txt](https://github.com/grpc/grpc/blob/master/CMakeLists.txt) is generated from a [template](https://github.com/grpc/grpc/blob/master/templates/CMakeLists.txt.template) and data from [build.yaml](https://github.com/grpc/grpc/blob/master/build.yaml).
Run tools/buildgen/generate_projects.sh to regenerate CMakelists.txt after updating build.yaml and/or the template. 
buil.yaml syntax is documented [here](https://github.com/grpc/grpc/blob/master/templates/README.md).

Reply:
I experimented build.yaml and tried to include everything into lib grpc++_cronet. Since not familiar with the syntax, still have several issues. You mentioned that you have made the change to package all codes into libgrpc_cronet. Looks like it's not committed in this PR yet. Could you give me the link of your change so I can try on that? thanks.

Reply:
Sorry, I had forgotten to push my change. I've updated the PR now. I'm now packaging all the code in libgrpc++_cronet.

Reply:
This single libgrpc++_cronet library works. Thanks! 

btw, we also explored another approach with build config. We added cornet transport and credentials source code to libgrpc and libgrpc++. It works too by just linking with these updated libgrpc & liggrpc++ only.